test/src/unit-lru_cache.cc

```diff
@@ -224,3 +225,35 @@ TEST_CASE_METHOD(
   auto it_end = lru_cache_->item_iter_end();
   CHECK(it == it_end);
 }
+
+TEST_CASE("BufferLRUCache of 0 capacity", "[lru_cache]") {
+  // Create 0 size cache
+  BufferLRUCache* lru_cache_ = new BufferLRUCache(CACHE_ZERO_SIZE);
+  auto it = lru_cache_->item_iter_begin();
+  auto it_end = lru_cache_->item_iter_end();
+  CHECK(it == it_end);
+
+  // Test insert
+  Buffer v;
+  CHECK(v.realloc(CACHE_ZERO_SIZE + 1).ok());
+  Status st = lru_cache_->insert("key", std::move(v));
+  CHECK(st.ok());
+
+  // Test read
+  Buffer v_buf;
+  bool success;
+  st = lru_cache_->read("key", &v_buf, 0, sizeof(int), &success);
+  CHECK(st.ok());
+  CHECK(!success);
+
+  // Test invalidate
+  st = lru_cache_->invalidate("key", &success);
+  CHECK(st.ok());
+  CHECK(!success);
+
+  // Test clear
+  lru_cache_->clear();
+  it = lru_cache_->item_iter_begin();
+  it_end = lru_cache_->item_iter_end();
+  CHECK(it == it_end);
+}
```

Comment:
The `lru_cache_` leaks here, please `delete` it (or use the `BufferLRUCacheFx` which will free its `lru_cache_` in its destructor).

Reply:
good catch, thanks!